### ISSUE TITLE
Obsolete diagnostic not reported on System.Index or System.Range

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#57745]
```

        [Fact]
        public void ObsoleteRangeType()
        {
            var source = @"
_ = new C()[..];

class C
{
    public int Length => 0;
    public int this[int i] => 0;
    public int Slice(int i, int j) => 0;
}

namespace System
{
    [Obsolete]
    public readonly struct Range
    {
        public Index Start { get; }

        public Index End { get; }

        public Range(Index start, Index end) => throw null;

        public static Range StartAt(Index start) => throw null;

        public static Range EndAt(Index end) => throw null;

        public static Range All => throw null;
    }
}
";
            // Note: we currently don't report Obsolete diagnostic on either Index or Range
            var comp = CreateCompilation(new[] { source, TestSources.Index });
            comp.VerifyDiagnostics();
        }
```

**Comment by Youssef1313** [comment]
This should have fallen off of https://github.com/dotnet/roslyn/pull/53463

**Comment by jcouv** [comment]
We discussed this topic earlier today with Jared/Aleksey/Fred and I wanted to record the team's stance: we're okay not to check Obsolete, UnmanagedCallersOnly or unexpected constraints on well-known types/members. I'm in process of [documenting](https://github.com/dotnet/roslyn/pull/75522) that.
Thanks @DoctorKrolic for looking into the issue. I'll go ahead and close it
